### PR TITLE
Don't load CLI extensions if database is missing or out of date

### LINKED ISSUE #20131: Extensions throw warning on migration when directus_extensions doesn't exist yet
### Describe the Bug

When migrating from < 10.7 to 10.7+, you might get a 

```
[11:24:12.030] WARN: Couldn't load extensions
[11:24:12.030] WARN: Cannot read properties of undefined (reading 'primary')
    err: {
      "type": "TypeError",
      "message": "Cannot read properties of undefined (reading 'primary')",
      "stack":
          TypeError: Cannot read properties of undefined (reading 'primary')
              at getASTFromQuery (/Users/rijkvanzanten/Developer/directus/directus/api/src/utils/get-ast-from-query.ts:79:51)
              at ItemsService.readByQuery (/Users/rijkvanzanten/Developer/directus/directus/api/src/services/items.ts:414:19)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at getExtensionsSettings (/Users/rijkvanzanten/Developer/directus/directus/api/src/extensions/lib/get-extensions-settings.ts:20:19)
              at ExtensionManager.load (/Users/rijkvanzanten/Developer/directus/directus/api/src/extensions/manager.ts:180:30)
              at ExtensionManager.initialize (/Users/rijkvanzanten/Developer/directus/directus/api/src/extensions/manager.ts:160:4)
              at createCli (/Users/rijkvanzanten/Developer/directus/directus/api/src/cli/index.ts:25:3)
    }
```

warning. It importantly doesn't prevent the migration, or prevents extensions from used once the API is started, but it looks scary!

### To Reproduce

Migrate from <10.7 to 10.7.0 with an existing database

### Directus Version

v10.7.0

### Hosting Strategy

Self-Hosted (Docker Image)

### LINKED ISSUE #20166: Fix `init` not running without DB

### PR DESCRIPTION
**Pull request by rijkvanzanten**
## Scope

What's changed:

- Warn instead of crash if extensions couldn't be loaded

## Potential Risks / Drawbacks

- None

## Review Notes / Questions

- None

---

Fixes #20166, fixes #20131

### CODE REVIEW

**Review by paescuj — CHANGES_REQUESTED** [review]
Still getting:
```console
[18:50:18.439] WARN: Couldn't load extensions
[18:50:18.439] WARN: Cannot read properties of undefined (reading 'primary')
    err: {
      "type": "TypeError",
      "message": "Cannot read properties of undefined (reading 'primary')",
      "stack":
          TypeError: Cannot read properties of undefined (reading 'primary')
              at getASTFromQuery (/Users/pascal/Development/directus/directus-5/api/src/utils/get-ast-from-query.ts:79:51)
              at ItemsService.readByQuery (/Users/pascal/Development/directus/directus-5/api/src/services/items.ts:414:19)
              at getExtensionsSettings (/Users/pascal/Development/directus/directus-5/api/src/extensions/lib/get-extensions-settings.ts:20:19)
              at ExtensionManager.load (/Users/pascal/Development/directus/directus-5/api/src/extensions/manager.ts:180:30)
              at ExtensionManager.initialize (/Users/pascal/Development/directus/directus-5/api/src/extensions/manager.ts:160:4)
              at createCli (/Users/pascal/Development/directus/directus-5/api/src/cli/index.ts:27:4)
    }
[18:50:18.490] INFO: Loaded extensions: test
```

**Comment by br41nslug** [comment]
> Still getting:
> 
> ```
> [18:50:18.439] WARN: Couldn't load extensions
> [18:50:18.439] WARN: Cannot read properties of undefined (reading 'primary')
>     err: {
>       "type": "TypeError",
>       "message": "Cannot read properties of undefined (reading 'primary')",
>       "stack":
>           TypeError: Cannot read properties of undefined (reading 'primary')
>               at getASTFromQuery (/Users/pascal/Development/directus/directus-5/api/src/utils/get-ast-from-query.ts:79:51)
>               at ItemsService.readByQuery (/Users/pascal/Development/directus/directus-5/api/src/services/items.ts:414:19)
>               at getExtensionsSettings (/Users/pascal/Development/directus/directus-5/api/src/extensions/lib/get-extensions-settings.ts:20:19)
>               at ExtensionManager.load (/Users/pascal/Development/directus/directus-5/api/src/extensions/manager.ts:180:30)
>               at ExtensionManager.initialize (/Users/pascal/Development/directus/directus-5/api/src/extensions/manager.ts:160:4)
>               at createCli (/Users/pascal/Development/directus/directus-5/api/src/cli/index.ts:27:4)
>     }
> [18:50:18.490] INFO: Loaded extensions: test
> ```

Is that coming from the `log.trace(error)` in the catch perhaps?

**Comment by paescuj** [comment]
> > Still getting:
> > ```
> > [18:50:18.439] WARN: Couldn't load extensions
> > [18:50:18.439] WARN: Cannot read properties of undefined (reading 'primary')
> >     err: {
> >       "type": "TypeError",
> >       "message": "Cannot read properties of undefined (reading 'primary')",
> >       "stack":
> >           TypeError: Cannot read properties of undefined (reading 'primary')
> >               at getASTFromQuery (/Users/pascal/Development/directus/directus-5/api/src/utils/get-ast-from-query.ts:79:51)
> >               at ItemsService.readByQuery (/Users/pascal/Development/directus/directus-5/api/src/services/items.ts:414:19)
> >               at getExtensionsSettings (/Users/pascal/Development/directus/directus-5/api/src/extensions/lib/get-extensions-settings.ts:20:19)
> >               at ExtensionManager.load (/Users/pascal/Development/directus/directus-5/api/src/extensions/manager.ts:180:30)
> >               at ExtensionManager.initialize (/Users/pascal/Development/directus/directus-5/api/src/extensions/manager.ts:160:4)
> >               at createCli (/Users/pascal/Development/directus/directus-5/api/src/cli/index.ts:27:4)
> >     }
> > [18:50:18.490] INFO: Loaded extensions: test
> > ```
> 
> ~Is that coming from the `log.trace(error)` in the catch perhaps?~ then it would not be a warning i suppose

Nope, the `catch` isn't reached

**Comment by paescuj** [comment]
I guess it's because the loading is already caught in the extension manager itself

**Review by br41nslug — DISMISSED** [review]

File: api/src/cli/load-extensions.ts [hunk 1]
```diff
@@ -12,10 +12,10 @@ export const loadExtensions = async () => {
 
 	if (!installed) return;
 
-	try {
-		await validateMigrations();
-	} catch {
-		logger.info('Database is out of date. Skipping CLI extensions initialization.');
+	const migrationsValid = await validateMigrations();
```
Comment (br41nslug) [inline api/src/cli/load-extensions.ts:15]:
```suggestion
	const migrationsComplete = await validateMigrations();
```
I'd probably call this `migrationsComplete` instead as we're not actually checking the migrations for validity just if have have all been run or not.

**Review by paescuj — APPROVED** [review]

**Pull request merged**